plugins/crypto/openssl/securitypolicy_eccnistp256.c

```diff
@@ -0,0 +1,722 @@
+                                           const UA_ByteString *nonce,
+                                           const UA_KeyValueMap *params,
+                                           UA_ByteString *csr,
+                                           UA_ByteString *newPrivateKey) {
+    /* Check parameter */
+    if(!securityPolicy || !csr)
+        return UA_STATUSCODE_BADINVALIDARGUMENT;
+
+     if(!securityPolicy->policyContext)
+         return UA_STATUSCODE_BADINTERNALERROR;
+
+    Policy_Context_EccNistP256 *pc =
+        (Policy_Context_EccNistP256*)securityPolicy->policyContext;
+
+    return UA_OpenSSL_CreateSigningRequest(pc->localPrivateKey, &pc->csrLocalPrivateKey,
+                                           securityPolicy, subjectName,
+                                           nonce, csr, newPrivateKey);
+}
+
+
+static UA_StatusCode
+updateCertificateAndPrivateKey_sp_EccNistP256(UA_SecurityPolicy *securityPolicy,
+                                                      const UA_ByteString newCertificate,
+                                                      const UA_ByteString newPrivateKey) {
+    if(securityPolicy == NULL)
+        return UA_STATUSCODE_BADINTERNALERROR;
+
+    if(securityPolicy->policyContext == NULL)
+        return UA_STATUSCODE_BADINTERNALERROR;
+
+    Policy_Context_EccNistP256 *pc =
+        (Policy_Context_EccNistP256 *)securityPolicy->policyContext;
+
+    UA_ByteString_clear(&securityPolicy->localCertificate);
+
+    UA_StatusCode retval = UA_OpenSSL_LoadLocalCertificate(
+        &newCertificate, &securityPolicy->localCertificate);
+
+    if(retval != UA_STATUSCODE_GOOD)
+        return retval;
+
+    /* Set the new private key */
+    EVP_PKEY_free(pc->localPrivateKey);
+
+    pc->localPrivateKey = UA_OpenSSL_LoadPrivateKey(&newPrivateKey);
+
+    if(!pc->localPrivateKey) {
+        retval = UA_STATUSCODE_BADSECURITYCHECKSFAILED;
+        goto error;
+    }
+
+    UA_ByteString_clear(&pc->localCertThumbprint);
+
+    retval = UA_Openssl_X509_GetCertificateThumbprint(&securityPolicy->localCertificate,
+                                                      &pc->localCertThumbprint, true);
+    if(retval != UA_STATUSCODE_GOOD) {
+        goto error;
+    }
+
+    return retval;
+
+error:
+    UA_LOG_ERROR(securityPolicy->logger, UA_LOGCATEGORY_SECURITYPOLICY,
+                 "Could not update certificate and private key");
+    if(securityPolicy->policyContext != NULL)
+        UA_Policy_EccNistP256_Clear_Context(securityPolicy);
+    return retval;
+}
+
+/* create the channel context */
+
+static UA_StatusCode
+UA_ChannelModule_EccNistP256_New_Context(const UA_SecurityPolicy *securityPolicy,
+                                                 const UA_ByteString *remoteCertificate,
+                                                 void **channelContext) {
+    if(securityPolicy == NULL || remoteCertificate == NULL || channelContext == NULL) {
+        return UA_STATUSCODE_BADINTERNALERROR;
+    }
```

Comment:
Use UA_calloc.
Then the memory is already initialized to zero.
And you can remove the _init.